openmdao/approximation_schemes/approximation_scheme.py

```diff
@@ -296,6 +297,16 @@ def _init_approximations(self, system):
     def _compute_approximations(self, system, jac, total, under_cs):
         from openmdao.core.component import Component
 
+        if hasattr(system._approx_schemes, "out_stream"):
```

Comment:
I think we can do this with 1 attribute instead of 2. Something like `self._progress_out`, which is either None (do nothing), or a stream (print the progress to the stream.)

Reply:
Note, this also needs to be declared in the __init__ , with default value None.  Also, needs to be in the class docstring.

Reply:
I think I see what you did. system._approximation_schemes is a dictionary that contains this approximation scheme. That is a roundabout way to do this, and I think the class attribute is a better way. You just need to know that `system._approximation_schemes` will be a dictionary that contains a single approximation scheme with the key either "fd" or "cs" depending on what was selected.